### DISCUSSION TITLE
Component Type Classification for a Runtime

### DISCUSSION THREAD

**Discussion by gilday** [CycloneDX/specification#139] [Q&A] [answered]
Which component type might best describe a program language runtime such as Java's runtime environment or Python's interpreter?

Maybe `application`, because are technically applications whose input is another application to run. There is some precedence for this: the Eclipse Temurin project describes their software as an application in their SBOM (see https://github.com/adoptium/temurin-build/pull/2869).

Maybe `framework`, because these runtimes fit the description of a software framework as defined in https://en.wikipedia.org/wiki/Software_framework.

> In [computer programming](https://en.wikipedia.org/wiki/Computer_programming), a software framework is an [abstraction](https://en.wikipedia.org/wiki/Abstraction_(computer_science)) in which [software](https://en.wikipedia.org/wiki/Software), providing generic functionality, can be selectively changed by additional user-written code, thus providing application-specific software. It provides a standard way to build and deploy applications and is a universal, reusable [software environment](https://en.wikipedia.org/wiki/Software_environment_(disambiguation)) that provides particular functionality as part of a larger [software platform](https://en.wikipedia.org/wiki/Software_platform) to facilitate the development of [software applications](https://en.wikipedia.org/wiki/Software_application), products and solutions.

**Comment by stevespringett** [comment, marked as answer]
Either application or framework may be used for now.  In the next version of the standard, we will likely introduce "platform" or similar term, to describe both local and SaaS runtime environments.

**Reply by gilday** [reply]
Thanks @stevespringett. By "next version of the standard", are you referring to the 1.5 or 2.0 milestone?

**Reply by stevespringett** [reply]
It's targeted at v1.5.
